### PR TITLE
Add some direct screen buffer writer methods to SSD13xx

### PR DESCRIPTION
**Pull request by SandorDobos**
## Description
I added some methods which allow faster display of bitmaps and fast partial screen clearing.

All existing ```Draw*()``` methods (including ```DrawBitmap()```)  use the ```DrawPixel()``` method 
to write anything into internal buffer which going to be transfered to device on ```Display()``` call.
Having bitmaps on hand this method is very slow.

Assuming the bitmaps are in appropriate format (see SSD13xx docs for "GDDRAM" and "Horizontal addressing mode")
the new ```DrawDirectAligned``` method simply copies the incoming byte array to appropriate place in internal buffer
which results is ~100 times faster display speed.
There are constraints: no bit operations occure with existing buffer data, so the "y" coordinate and the bitmap height must be byte aligned!

Same constraints apply to new ```ClearDirectAligned``` method which allow partial (rectangle) screen clearing 
by setting appropriate internal buffer bytes to 0x00. This is also ~100 times faster than doing same with existing ```DrawFilledRectangle``` method.
 

## Motivation and Context
Using bitmaps actively the existing methods were unacceptable slow.

## How Has This Been Tested?
I am using this implementation in my own project.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (work on Unit Tests, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: devices/Ssd13xx/Ssd13xx.cs [hunk 1]
```diff
@@ -145,6 +145,72 @@ protected SpanByte SliceGenericBuffer(int start, int length)
...
+
+            var dataHeightInBytes = height / 8;
+            if ((dataHeightInBytes * width) != buffer.Length)
+            {
+                throw new ArgumentException("Width and height do not match the bitmap size.");
+            }
+
+            var genericBufferIdx = y / 8 * Width + x;
+            var srcIdx = 0;
+            for (int i=0; i<dataHeightInBytes; i++)
```
Comment (Ellerbach) [inline devices/Ssd13xx/Ssd13xx.cs:177]: please use `ctrl + k + d` in Visual Studio to have a proper alignment of the code ;-)

File: devices/Ssd13xx/Ssd13xx.cs [hunk 2]
```diff
@@ -145,6 +145,72 @@ protected SpanByte SliceGenericBuffer(int start, int length)
...
+        /// <summary>
+        /// Copies buffer content directly to display buffer.
+        /// Y and height must be byte aligned because buffer will 
+        /// be copied without any logical operations on existing content.
+        /// </summary>
+        /// <param name="x">The x coordinate on the screen.</param>
+        /// <param name="y">The y coordinate on the screen.</param>
+        /// <param name="width">Width of buffer content in pixels.</param>
+        /// <param name="height">Height of buffer content in pixels.</param>
+        /// <param name="buffer">Data to copy.</param>
```
Comment (Ellerbach) [inline devices/Ssd13xx/Ssd13xx.cs:157]: please with the size the buffer needs to have to make it clear. In short, height * width / 8.

**Review by Ellerbach — COMMENTED** [review]
Few comments. Looks good!
Can you add as well a code snippet in the Readme or in the sample on how to use the 2 functions.

**Comment by josesimoes** [comment]
@SandorDobos can you please address @Ellerbach  comment above so we can have this merged?

**Comment by SandorDobos** [comment]
@josesimoes , @Ellerbach Done :)

**Review by josesimoes — CHANGES_REQUESTED** [review]

File: devices/Ssd13xx/README.md [hunk 3]
```diff
@@ -27,3 +27,49 @@ The following connection types are supported by this binding.
 
 - [X] I2C
 - [ ] SPI
+
+## Performance suggestions
+
+If you wish to update big part of screen (e.g. drawing images or shapes instead of printing text)
```
[the remark this answers is not in the document]
Comment (SandorDobos) [reply devices/Ssd13xx/README.md]: I would say that the lines above line 40 contain importand developer information about how other methods work (they are just functional wrappers around ````DrawPixel````). This information has a value without the new methods and their explanation. I reworded the whole section, hope you will like this version :)

[the remark this answers is not in the document]
Comment (josesimoes) [reply devices/Ssd13xx/README.md]: From a developer looking into the inner bits of the library or trying to understand deeper, I agree. For a "regular" _just_ wanting to use the library, those are spurious.

That's why they belong in the PR or in the code comments. To help on those situation and for developers diving deep in the code.

It's better now. Thanks!

**Pull request merged**